oops.min.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-var */
+/* eslint-disable max-len */
+
+// eslint-disable-next-line require-jsdoc
+function Oops(a) {
+  const c=this; const d={defaultConfig: {storage: 'localStorage', key: 'oops_key'}, config: {}};
+  var b={isUndefined: function(e) {
+    return e===undefined||e===null;
+  }, isDefined: function(e) {
+    return e!==undefined&&e!==null;
+  }, def: function(g, f, h, e) {
+    Object.defineProperty(g, f, {value: h, enumerable: !!e, writable: true, configurable: true});
+  }, merge: function(g) {
+    const e={}; for (const f in d.defaultConfig) {
+      if (d.defaultConfig.hasOwnProperty(f)) {
+        if (g.hasOwnProperty(f)) {
+          b.def(e, f, g[f], true);
+        } else {
+          b.def(e, f, d.defaultConfig[f], true);
+        }
+      }
+    } return e;
+  }, storageAvailable: function(h) {
+    let g;
+    try {
+      g=window[h]; const i='__storage_test__'; g.setItem(i, i); g.removeItem(i); return true;
+    } catch (f) {
+      return f instanceof DOMException&&(f.code===22||f.code===1014||f.name==='QuotaExceededError'||f.name==='NS_ERROR_DOM_QUOTA_REACHED')&&(g&&g.length!==0);
+    }
+  }, storageSetItems: function(e) {
+    switch (d.config.storage) {
+      case 'localStorage': localStorage.setItem(d.config.key, JSON.stringify(e));
+        break; case 'sessionStorage': sessionStorage.setItem(d.config.key, JSON.stringify(e));
+        break;
+    }
+  }, storageGetItems: function() {
+    switch (d.config.storage) {
+      case 'localStorage': return JSON.parse(localStorage.getItem(d.config.key));
+      case 'sessionStorage': return JSON.parse(sessionStorage.getItem(d.config.key));
+    }
+  }, isset: function() {
+    return (b.isDefined(sessionStorage.getItem(d.config.key))||b.isDefined(localStorage.getItem(d.config.key)));
+  }, prepare: function(e) {
+    const f={}; e.forEach(function(h, g) {
+      if ('type' in h) {
+        b.def(f, g, h.value, true);
+      }
+    }); b.def(f, 'selected', e.length, true); return f;
+  }, noError: function(e) {
+    if (e=='localStorage'||e=='sessionStorage') {
+      if (b.storageAvailable(e)) {
+        return true;
+      } else {
+        console.warn('oops.js warn: '+e+' is not available.'); return false;
```

Comment:
- [ ] For best practice please consider not committing  ```console.warn``` or ```console.log``` to your repo it makes the code looks messy.